src/containerConfig.ts

```diff
@@ -24,25 +24,35 @@ async function registerExternalValues(): Promise<void> {
   const tracer = trace.getTracer(SERVICE_NAME);
   container.register(SERVICES.TRACER, { useValue: tracer });
   container.register(SERVICES.LOGGER, { useValue: logger });
-  const redisConnection = await createConnection(config.get<RedisOptions>('db.connection.options'));
-  redisConnection.on('error', (err) => {
-    logger.fatal(err, `Redis connection failure, exiting with code ${REDIS_CONNECTION_ERROR_CODE}`);
-    return process.exit(REDIS_CONNECTION_ERROR_CODE);
-  });
-
-  container.register(REDIS_SYMBOL, { useValue: redisConnection });
-  container.register(IDOMAIN_FIELDS_REPO_SYMBOL, { useClass: RedisManager });
 
   const schemas = await getSchemas(container);
   container.register(schemaSymbol, { useValue: schemas });
 
+  const connectToExternal = schemas.some((schema) => {
+    return schema.enableExternalFetch === 'yes';
+  });
+
+  let redisConnection: Redis;
+
+  if (connectToExternal) {
+    redisConnection = createConnection(config.get<RedisOptions>('db.connection.options'));
+    redisConnection.on('error', (err) => {
+      logger.fatal(err, `Redis connection failure, exiting with code ${REDIS_CONNECTION_ERROR_CODE}`);
+      return process.exit(REDIS_CONNECTION_ERROR_CODE);
+    });
+    container.register(REDIS_SYMBOL, { useValue: redisConnection });
+    container.register(IDOMAIN_FIELDS_REPO_SYMBOL, { useClass: RedisManager });
+  } else {
+    container.register(IDOMAIN_FIELDS_REPO_SYMBOL, { useValue: {} });
+  }
```

Comment:
I don't think redis connection should depend on any schema using it. If we want the option to turn it off, it should be a configuration option. WDYT @syncush ?